### PR TITLE
Add a nightly hotpatch utils for python only PR

### PR DESCRIPTION
**Pull request by bhack**
I think this could help many teams, especially compile/export teams (/cc @ezyang), to let end user/bug reporters to quickly test WIP PR when reporting a related bug.

This could quickly run in an official nightly Docker container or in  a nightly venv/coda env.

Let me know what do you think.

### CODE REVIEW

**Comment by bhack** [comment]
We can eventually expand this in another PR for supporting a chain of PRs as the nightly gap vs main could eventually require this type of patching

**Comment by ezyang** [comment]
I don't have a fundamental objection to this code existing but I wonder if it is better off just living somewhere, e.g., in a gist and linking people to it.

**Comment by bhack** [comment]
The advantage eventually is to have it off the shelf in the container with any extra download. Also I suppose a gist will be lost if you need to propose it time by time to users on single issues

File: tools/nightly_hotpatch.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import sys
+import os
+import subprocess
+import urllib.request
+import tempfile
+import shutil
+from pathlib import Path
+
+def print_usage():
+    print(f"Usage: {sys.argv[0]} <PR_NUMBER>")
```
Comment (ezyang) [inline tools/nightly_hotpatch.py:12]: Please write a proper help, there are literally no docs anywhere haha

Comment (bhack) [reply tools/nightly_hotpatch.py]: I've added an help with `argparse`

**Comment by bhack** [comment]
I don't know if we want to add something in this section
https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#nightly-checkout--pull

But I think the main scope of this new script is for triagers or core team members to eventually request this action to users to  under review or intra-nightlies merged PRs for user specific tickets.

**Review by ezyang — APPROVED** [review]
alright, seems harmless enough, let's see if there's any uptake

**Comment by bhack** [comment]
Can help socialize this internally? 
It could be nice to collect feedback on real use cases.

**Comment by ezyang** [comment]
unfortunately, there's not much use case at Meta haha, everyone's running on main LOL

**Comment by bhack** [comment]
> unfortunately, there's not much use case at Meta haha, everyone's running on main LOL

It is not for Meta is for external user. Meta and core team members often ask to apply PR to test if an issue is merged especially in the compiler stack, export, etc..

**Pull request closed without merging**